pkg/dependencies/factory.go

```diff
@@ -43,15 +42,14 @@ type Dependencies struct {
 	DignosticCollectorFactory diagnostics.DiagnosticBundleFactory
 	CAPIManager               *clusterapi.Manager
 	ResourceSetManager        *clusterapi.ResourceSetManager
+	closers                   []types.Closer
 }
 
 func (d *Dependencies) Close(ctx context.Context) error {
```

Comment:
Good point
I went with `Close` for all the "cleanup" methods since that's the "standard" way (not really a standard, just the most common one I think) to write "destructors" (not really a destructor since it's not called when the gc frees the object)

That said, I also like the idea of having the same signature for all the cleanup methods. It allows us to treat everything as a `Closer` like in `func close(ctx context.Context, closer types.Closer)`